### PR TITLE
New AdvSec & GHAS - Enable using config file

### PR DESCRIPTION
**Pull request by shaopeng-gh**
@EasyRhinoMSFT 
This is to enable using config file to change settings for New AdvSec & GHAS work.

### CODE REVIEW

File: src/Sarif/RuleKindSet.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    [Serializable]
+    [JsonConverter(typeof(TypedPropertiesDictionaryConverter))]
+    public class RuleKindSet : HashSet<RuleKind>
```
Comment (EasyRhinoMSFT) [inline src/Sarif/RuleKindSet.cs:14]: >RuleKindSet [](http://example.com/codeflow?start=17&length=11)

Why couldn't we use a HashSet?

File: src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs [hunk 2]
```diff
@@ -319,7 +319,7 @@ public virtual TContext InitializeGlobalContextFromOptions(TOptions options, ref
                 context.TargetFileSpecifiers = options.TargetFileSpecifiers?.Any() == true ? InitializeStringSet(options.TargetFileSpecifiers) : context.TargetFileSpecifiers;
                 context.InvocationPropertiesToLog = options.InvocationPropertiesToLog?.Any() == true ? InitializeStringSet(options.InvocationPropertiesToLog) : context.InvocationPropertiesToLog;
                 context.Traces = options.Trace.Any() ? InitializeStringSet(options.Trace) : context.Traces;
-                context.RuleKinds = options.RuleKinds;
+                context.RuleKinds = options.RuleKindOption != null ? options.RuleKinds : context.RuleKinds;
```
Comment (EasyRhinoMSFT) [inline src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs:322]: >options.RuleKindOption != null ? options.RuleKinds : context.RuleKinds [](http://example.com/codeflow?start=36&length=70)

Can we use this simpler syntax?

options.RuleKindOption ?? context.RuleKinds

Comment (shaopeng-gh) [reply src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs:322]: no we can not,
options.RuleKindOption is not the same type as options.RuleKinds

File: src/Sarif/RuleKindSet.cs (hunk 1 above)
Comment (shaopeng-gh) [reply src/Sarif/RuleKindSet.cs:14]: this is following the same pattern as other fields we have, the ResultKind.
The commanline nuget package the options have some limitation so there is some conversions needed

File: src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs (hunk 2 above)
Comment (shaopeng-gh) [reply src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs:322]: if you see lines above
`context.ResultKinds = options.Kind != null ? options.ResultKinds : context.ResultKinds;`
Some tricky here, the `options.Kind` is not the same type as `options.ResultKinds`.

p.s. I think the naming before is not good for `Kind` for `ResultKinds` , it is better to name using xxxOption, or some pattern. People will not know `Kind` is the command line option for `ResultKinds`

**Review by EasyRhinoMSFT — APPROVED** [review]
:shipit:

**Pull request merged**